vscode-trace-extension/src/external-api/external-api.ts

```diff
@@ -0,0 +1,43 @@
+/***************************************************************************************
+ * Copyright (c) 2023 BlackBerry Limited and others.
+ *
+ * Licensed under the MIT license. See LICENSE file in the project root for details.
+ ***************************************************************************************/
+import { Experiment } from 'tsp-typescript-client/lib/models/experiment';
+import { TraceViewerPanel } from '../trace-viewer-panel/trace-viewer-webview-panel';
+import { traceExtensionEventManager } from '../extension';
+
+export interface ExternalAPI {
+    getActiveExperiment(): Experiment | undefined,
+    fireEventEmitter(eventId: string, data: unknown): void,
+    getEventEmitterIds(): Iterable<string>
+}
+
+export const traceExtensionAPI: ExternalAPI = {
+    /**
+     * Retrieves the currently active experiement
+     *
+     * @returns Experiment if one is currently active, otherwise undefined
+     */
+    getActiveExperiment(): Experiment | undefined {
+        return TraceViewerPanel.getCurrentExperiment();
+    },
+
+    /**
+     * Fires the event emitter with the provided ID
+     *
+     * @param eventId ID of the event emitter to fire
+     * @param data event object
+     */
+    fireEventEmitter(eventId: string, data: unknown): void {
```

Comment:
It took me a while to understand how it's implemented. When an extension wants to be notified about a signal received by a webview, it uses this API to fire an event through the eventEmitter that will add a callback to a webview which is called when a signal is received by the webview and contains the signal name provided (when calling fireEventEmitter). I guess EventEmitter is the standard way to communicate between webview and that's why you used this approach. Or was that another reason. Please let me know. 

Second questions, should `data` be of type `SignalCallbackInterface`? Or is it set to `unknown` so that we can use if for other use cases than signal callbacks? Do you have one in mind?